### ISSUE TITLE
scopeRole not working

### ISSUE THREAD

**Issue by whfamily2006** [spatie/laravel-permission#191]
User Model:

```
class User extends Authenticatable implements AuthenticatableContract, CanResetPasswordContract {
	
	use Notifiable, RevisionableTrait, CanResetPassword, HasRoles, SoftDeletes;
	
	protected $softDelete               = true;
	protected $dates                    = [ 'deleted_at' ];
	protected $revisionCreationsEnabled = true;
	protected $dontKeepRevisionOf       = [ 'remember_token' ];
	protected $guarded                  = [];
	protected $hidden                   = [ 'password','remember_token' ];
}
```
Laravel-Permission config file: 

```
return [
    'models' => [
        'permission' => Spatie\Permission\Models\Permission::class,
        'role' => Spatie\Permission\Models\Role::class,
    ],

    'table_names' => [
        'users' => 'users',
        'roles' => 'roles',
        'permissions' => 'permissions',
        'user_has_permissions' => 'user_has_permissions',
        'user_has_roles' => 'assigned_roles',
        'role_has_permissions' => 'permission_role',

    ],

];

```

My Query: 

`return User::role( [ 'User', 'ResellerRemitter'] )->count();`

Error: 

```
QueryException in Connection.php line 770: 

SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'id' in where clause is ambiguous (SQL: select count(*) as aggregate from `users` where exists (select * from `roles` inner join `assigned_roles` on `roles`.`id` = `assigned_roles`.`role_id` where `assigned_roles`.`user_id` = `users`.`id` and (`id` = 16 or `id` = 18)) and `users`.`deleted_at` is null)
```
@Rias500 @freekmurze

**Comment by AlexVanderbist** [comment]
Hi, 

I couldn't replicate the issue even if I used exactly the same query. This might be because we're on different mysql versions (or different database drivers in general).

I did however submit a PR adding a table prefix to the only ambiguous part of the query (around `('id' = ? or 'id' = ?)`). This might fix the issue you're having.

**Comment by freekmurze** [comment]
Fix by @AlexVanderbist is tagged as `1.9.0`
